src/WebJobs.Extensions.CosmosDB/Trigger/CosmosDBTriggerAttributeBindingProviderGenerator.cs

```diff
@@ -0,0 +1,78 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Reflection;
+using System.Threading.Tasks;
+using Microsoft.Azure.WebJobs.Host.Triggers;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.Logging;
+using Newtonsoft.Json.Linq;
+
+namespace Microsoft.Azure.WebJobs.Extensions.CosmosDB
+{
+    /// <summary>
+    /// Detects the user defined type T on the binding and calls the <see cref="CosmosDBTriggerAttributeBindingProvider{T}"/>.
+    /// </summary>
+    internal class CosmosDBTriggerAttributeBindingProviderGenerator : ITriggerBindingProvider
+    {
+        private readonly IConfiguration _configuration;
+        private readonly INameResolver _nameResolver;
+        private readonly CosmosDBOptions _options;
+        private readonly ILoggerFactory _loggerFactory;
+        private readonly CosmosDBExtensionConfigProvider _configProvider;
+
+        public CosmosDBTriggerAttributeBindingProviderGenerator(IConfiguration configuration, INameResolver nameResolver, CosmosDBOptions options,
+            CosmosDBExtensionConfigProvider configProvider, ILoggerFactory loggerFactory)
+        {
+            _configuration = configuration;
+            _nameResolver = nameResolver;
+            _options = options;
+            _configProvider = configProvider;
+            _loggerFactory = loggerFactory;
+        }
+
+        public static Type GetParameterType(ParameterInfo parameter) => parameter.ParameterType.GenericTypeArguments.Length > 0 ? parameter.ParameterType.GenericTypeArguments[0] : parameter.ParameterType;
+
+        public Task<ITriggerBinding> TryCreateAsync(TriggerBindingProviderContext context)
+        {
+            CosmosDBTriggerAttribute cosmosDBTriggerAttribute = context.Parameter.GetCustomAttribute<CosmosDBTriggerAttribute>(inherit: false);
+
+            if (cosmosDBTriggerAttribute == null)
+            {
+                return Task.FromResult<ITriggerBinding>(null);
+            }
+
+            Type documentType = CosmosDBTriggerAttributeBindingProviderGenerator.GetParameterType(context.Parameter);
+
+            if (typeof(JArray).IsAssignableFrom(documentType))
```

Comment:
Oh right... I forgot we required a Newtonsoft ref...

Reply:
If we were to change to System.Text.Json, which would be the equivalent?